scripts/build_images.py

```diff
@@ -0,0 +1,257 @@
+import hashlib
+import json
+import os
+import subprocess
+import sys
+import tempfile
+
+# The container name for the local Docker registry.
+REGISTRY_NAME = "prairielearn-registry"
+
+# A mapping from images to their base images. We use this to know to rebuild images when their base image has changed.
+BASE_IMAGE_MAPPING = {
+    # Core images.
+    "prairielearn/prairielearn": "prairielearn/plbase",
+    "prairielearn/executor": "prairielearn/prairielearn",
+    # Workspace images.
+    "prairielearn/workspace-vscode-python": "prairielearn/workspace-vscode-base",
+    "prairielearn/workspace-vscode-cpp": "prairielearn/workspace-vscode-base",
+}
```

Comment:
I briefly tried to avoid this, but I don't think it's terrible to have to maintain this little dict in the long run.